plotly_resampler/aggregation/aggregation_interface.py

```diff
@@ -4,169 +4,237 @@
+        y: np.ndarray = None,
+        n_out: int = None,
+        **kwargs,
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Aggregate the data.
 
-        return med_diff, s_idx_diff
+        Parameters
+        ----------
+        x : np.ndarray, optional
+            The x-axis data, by default None
+        y : np.ndarray
+            The y-axis data
+        n_out : int, optional
+            The number of output datapoints, by default None
+        kwargs : dict
+            Additional keyword arguments
 
-    def _insert_gap_none(self, s: pd.Series) -> pd.Series:
-        # ------- INSERT None between gaps / irregularly sampled data -------
-        med_diff, s_idx_diff = self._calc_med_diff(s)
-        # add None data-points in-between the gaps
-        if med_diff is not None:
-            df_gap_idx = s.index.values[s_idx_diff > 3 * med_diff]
-            if len(df_gap_idx):
-                df_res_gap = pd.Series(
-                    index=df_gap_idx, data=None, name=s.name, copy=False, dtype=s.dtype
-                )
-
-                if isinstance(df_res_gap.index, pd.DatetimeIndex):
-                    # Due to the s.index`.values` cast, df_res_gap has lost
-                    # time-information, so now we restore it
-                    df_res_gap.index = df_res_gap.index.tz_localize("UTC").tz_convert(
-                        s.index.tz
-                    )
-
-                # Note:
-                #  * the order of pd.concat is important for correct visualization
-                #  * we also need a stable algorithm for sorting, i.e., the equal-index
-                #    data-entries their order will be maintained.
-                s = pd.concat([df_res_gap, s], ignore_index=False).sort_index(
-                    kind="mergesort"
-                )
-        return s
-
-    def _replace_gap_end_none(self, s: pd.Series) -> pd.Series:
-        # ------- REPLACE None where a gap ends -------
-        med_diff, s_idx_diff = self._calc_med_diff(s)
-        if med_diff is not None:
-            # Replace data-points with None where the gaps occur
-            # The default is the end of a gap
-            nan_mask = s_idx_diff > 4 * med_diff
-            if self.nan_position == "begin":
-                # Replace the last non-gap datapoint (begin of gap) with Nan
-                nan_mask = np.roll(nan_mask, -1)
-            elif self.nan_position == "both":
-                # Replace the encompassing gap datapoints with Nan
-                nan_mask |= np.roll(nan_mask, -1)
-            s.loc[nan_mask] = None
-        return s
-
-    def aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
-        """Aggregate (downsample) the given input series to the given n_out samples.
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The aggregated x and y data, respectively
+        """
+        assert n_out is not None
+
+        # assert isinstance(y, np.ndarray)
```

Comment:
When we uncomment this, the tests fail...

Reply:
Let's add a TODO here then to clearly indicate that we need to check that. Also worth including our best theory as to why that might be failing.

Reply:
Just found out why - we still use pandas.Categorical to represent category dtypes - this is what causes the test to fail in my rework in #186